easybuild/framework/easyblock.py

```diff
@@ -2720,20 +2720,72 @@ def prepare_step(self, start_dir=True, load_tc_deps_modules=True):
         if start_dir:
             self.guess_start_dir()
 
+    def _run_cmds(self, cmds):
+        """Execute list of shell commands"""
+        self.log.debug(f"List of commands to be executed: {cmds}")
+
+        # make sure we have a list of commands
+        if not isinstance(cmds, (list, tuple)):
+            raise EasyBuildError(
+                f"Invalid argument for EasyBlock._run_cmds, expected list or tuple of strings: {cmds}"
+            )
+
+        for cmd in cmds:
+            if not isinstance(cmd, str):
+                raise EasyBuildError(f"Invalid element in command list, not a string: {cmd}")
+            run_shell_cmd(cmd)
+
+    def run_core_step_cmd(self, step):
+        """Construct main command of core step and execute it"""
+
+        core_steps = ["configure", "build", "test", "install"]
+        if step not in core_steps:
+            raise EasyBuildError(
+                f"Cannot construct command for step {step}. Supported steps are: {', '.join(core_steps)}."
+            )
+
+        step_cmd_cfgs = {
+            "configure": ('pre_configure_cmds', 'preconfigopts', 'configure_cmd', 'configopts'),
+            "build": ('pre_build_cmds', 'prebuildopts', 'build_cmd', 'buildopts'),
+            "test": ('pre_test_cmds', 'pretestopts', 'test_cmd', 'testopts'),
+            "install": ('pre_install_cmds', 'preinstallopts', 'install_cmd', 'installopts'),
+        }
+
+        pre_step_cmds = self.cfg[step_cmd_cfgs[step][0]]
+        if pre_step_cmds:
+            try:
+                self._run_cmds(pre_step_cmds)
```

Comment:
Don't we limit the usefulness of `pre_configure_cmds` by running those commands separately?

Something like `export FOO=bar` in `pre_configure_cmds` won't work as expected because it's run in a different shell than `configure_cmd`, which limits the usefulness of `pre_configure_cmds` quite a bit I think?

I can see the value though (better errors when one of the commands in `pre_configure_cmds` fails, for example), but we should try and make it very clear that these commands are running in a different subshell?